apps/ejabberd/src/mod_muc_light.erl

```diff
@@ -203,9 +205,14 @@ route(From, To, Packet) ->
                      DecodedPacket :: mod_muc_light_codec:decode_result(),
                      OrigPacket :: jlib:xmlel()) -> any().
 process_packet(From, To, {ok, {set, #create{} = Create}}, OrigPacket) ->
-    RoomsPerUser = get_opt(To#jid.lserver, rooms_per_user, ?DEFAULT_ROOMS_PER_USER),
     FromUS = jid:to_lus(From),
-    case RoomsPerUser == infinity orelse length(?BACKEND:get_user_rooms(FromUS)) < RoomsPerUser of
+    MayCreate = case get_opt(To#jid.lserver, rooms_per_user, ?DEFAULT_ROOMS_PER_USER) of
+                    infinity ->
+                        true;
+                    RoomsPerUser ->
+                        length(?BACKEND:get_user_rooms(FromUS, To#jid.lserver)) < RoomsPerUser
```

Comment:
How difficult would it be to implement `?BACKEND:get_user_rooms_count`? With count on the backend side?